static/js/tools/shared/tile_metadata_modal.tsx

```diff
@@ -0,0 +1,138 @@
+ */
+
+import React, { useEffect, useState } from "react";
+import { Button, Modal, ModalBody, ModalFooter, ModalHeader } from "reactstrap";
+
+import { StatVarSpec } from "../../shared/types";
+import { datacommonsClient } from "../../utils/datacommons_client";
+
+const SV_EXPLORER_REDIRECT_PREFIX = "/tools/statvar#sv=";
+
+interface TileMetadataModalPropType {
+  statVarSpecs: StatVarSpec[];
+}
+
+type Dcid = string;
+type Name = string;
+type DcidNameTuple = [Dcid, Name];
+
+function buildMetadataRow(dcid: Dcid, name: Name, key: number): JSX.Element {
+  return (
+    <div className="metadata-modal-link" key={key}>
+      <span className="material-icons-outlined">arrow_forward</span>
+      <a
+        href={SV_EXPLORER_REDIRECT_PREFIX + dcid}
+        target="_blank"
+        rel="noreferrer"
+      >
+        {name}
+      </a>
+    </div>
+  );
+}
+
+export function TileMetadataModal(
+  props: TileMetadataModalPropType
+): JSX.Element {
+  const { statVarSpecs } = props;
+  const [modalOpen, setModalOpen] = useState(false);
+  const [statVarNames, setStatVarNames] = useState<DcidNameTuple[]>([]);
+  const dcids = new Set<string>();
+
+  if (statVarSpecs) {
+    for (const spec of statVarSpecs) {
+      dcids.add(spec.statVar);
+      if (spec.denom) {
+        dcids.add(spec.denom);
+      }
+    }
+  }
+
+  useEffect(() => {
+    // Only fetch data once the modal is opened.
+    if (!modalOpen) return;
+    if (dcids.size == statVarNames.length) return;
+    (async () => {
+      const responseObj = await datacommonsClient.getFirstNodeValues({
+        dcids: [...dcids],
+        prop: "name",
+      });
+      const responseList = new Array<DcidNameTuple>();
+      for (const dcid in responseObj) {
+        responseList.push([dcid, responseObj[dcid]]);
+      }
+      // Sort by name
+      responseList.sort((a, b) => (a[1] > b[1] ? 1 : -1));
+      setStatVarNames(responseList);
+    })();
+  }, [props, modalOpen, dcids, statVarNames.length]);
+
+  return (
+    <>
+      <a
+        href="#"
+        onClick={(e) => {
+          e.preventDefault();
+          setModalOpen(true);
+        }}
+      >
+        show metadata
+      </a>
+      {modalOpen && (
+        <Modal
+          isOpen={modalOpen}
+          scrollable
+          keyboard
+          className="metadata-modal modal-dialog-centered modal-lg"
+        >
+          <ModalHeader toggle={() => setModalOpen(false)} close={<></>}>
+            Choose a variable to view its metadata
+          </ModalHeader>
+          <div className="modal-subtitle">
+            Select a variable from the list to see its details. The links below
+            open the Statistical Variable Explorer in a new tab.
+          </div>
+          <ModalBody>
+            <div className="metadata-modal-links">
+              {statVarNames.length
+                ? statVarNames.map((dcidName, i) =>
+                    buildMetadataRow(dcidName[0], dcidName[1], i)
```

Comment:
alternative: `<MetadataRow dcid={dcidName[0]} name={dcidName[1]} key={i} />`

Reply:
done!

Reply:
Is there a missing commit that didn't get pushed? I don't think I see the update.

Reply:
pushed -- rookie mistake :)